### PR TITLE
feat(trends): Implicitly set trend parameter as filter value in query

### PR DESCRIPTION
**Pull request by udameli**
For the performance trends view, add current trend parameter field value as an additional condition filter to the query

### CODE REVIEW

File: static/app/views/performance/data.tsx [hunk 1]
```diff
@@ -456,6 +457,10 @@ function generateGenericPerformanceEventView(
     eventView.additionalConditions.addFilterValues('event.type', ['transaction']);
   }
 
+  if (isTrends) {
+    eventView.additionalConditions.addFilterValues('has', ['measurements.vital']);
+  }
```
Comment (udameli) [inline static/app/views/performance/data.tsx:462]: `has:measurements.vital` should only be added to the trends query (that was the assumption i made)

Comment (Zylphrex) [reply static/app/views/performance/data.tsx:462]: `measurements.vitals` is not an actual measurement. In this case, the measurement we want to add depends on the parameter dropdown. So if `LCP` was selected, you'll want to add `has:measurements.lcp` and so on. Try doing this manually in the search bar to see how that affects the results.

image

**Comment by k-fish** [comment]
Needs a few tweaks, discussed offline 😄

File: static/app/views/performance/data.tsx (hunk 1 above)
Comment (udameli) [reply static/app/views/performance/data.tsx:466]: sorry my initial understanding of the task wasn't accurate. Updated the approach

**Comment by udameli** [comment]
> Needs a few tweaks, discussed offline 😄

done!

**Review by k-fish — COMMENTED** [review]
Looks great 👏 , we should probably include a test here to make sure this implicit condition is making it to the api request.

I'd probably add or modify one of the tests in `tests/js/spec/views/performance/trends/index.spec.jsx`

File: tests/js/spec/views/performance/trends/index.spec.jsx [hunk 2]
```diff
@@ -410,6 +411,56 @@ describe('Performance > Trends', function () {
     }
   }, 10000);
 
+  it('choosing a web vitals parameter adds it as an additional condition to the query', async function () {
```
Comment (k-fish) [inline tests/js/spec/views/performance/trends/index.spec.jsx:414]: 👏

**Review by k-fish — APPROVED** [review]
Looks great, 🚢  it!

**Pull request merged**